plugins/obs-ffmpeg/obs-ffmpeg.c

```diff
@@ -190,6 +245,12 @@ static bool nvenc_supported(void)
 		goto finish;
 	}
 #else
+#ifdef __linux__
```

Comment:
I think this would be better with `#elif defined(__linux__)`

Reply:
I tried to be the least intrusive here as possible. For example - is NVENC on FreeBSD a thing? (I have seen people proposing FreeBSD patches as a platform here from time to time). So for that case I wanted to do the check only on Linux, but in the non-Linux but also non-Windows case still trying to `os_dlopen()` and continue if successful as we currently do for Linux for not breaking them. FreeBSD would need it's own set of checks I believe.

Reply:
Im not sure i understand the concern. Freebsd would need to implement separate checks which isnt affected by marking the exclusivity of the windows and linux checks with elif instead of two separate if blocks.

Reply:
I think it would duplicate code then.
```
if win32
..
elseif linux
  blacklist nvidia linux()
  lib = os_dlopen("libnvidia-encode.so.1");
else #freebsd
  lib = os_dlopen("libnvidia-encode.so.1");
endif
```
versus
```
if win32
..
else
  ifdef linux
    blacklist nvidia linux()
  endif
  lib = os_dlopen("libnvidia-encode.so.1");
endif
```
Personally I don't mind either way. It is a decision when and where to have code duplication or not.

Reply:
Hmm i missed this was nested within the else block of win32 (part of why id prefer it flatter). A little code duplication for simpler structure is my personal preference. Like the rest this is mostly a style choice if you prefer to leave it this way feel free.